src/internal/resources/manager.cpp

```diff
@@ -151,4 +177,38 @@ PartitionResources& Manager::partition(std::size_t partition_id)
     CHECK_LT(partition_id, m_partitions.size());
     return m_partitions.at(partition_id);
 }
+
+Manager& Manager::get_resources()
+{
+    if (m_thread_resources == nullptr)  // todo(cpp20) [[unlikely]]
+    {
+        LOG(ERROR) << "thread with id " << std::this_thread::get_id()
+                   << " attempting to access the SRF runtime, but is not a SRF runtime thread";
+        throw exceptions::SrfRuntimeError("can not access runtime resources from outside the runtime");
+    }
+
+    return *m_thread_resources;
+}
+
+PartitionResources& Manager::get_partition()
+{
+    {
+        if (m_thread_partition == nullptr)  // todo(cpp20) [[unlikely]]
+        {
+            auto& resources = Manager::get_resources();
+
+            if (resources.system().partitions().device_to_host_strategy() == PlacementResources::Shared)
+            {
+                LOG(ERROR) << "runtime partition query is disabed when PlacementResources::Shared is in use";
+                throw exceptions::SrfRuntimeError("unable to access runtime parition info");
+            }
+
+            LOG(ERROR) << "thread with id " << std::this_thread::get_id()
```

Comment:
If this is something that we really want to prevent; would it be worthwhile to store the thread id on the resource object, and compare it with the requesting thread, rather than this kind of implicit check?